### PR TITLE
Modularized setup guides

### PR DESCRIPTION
**Pull request by anniedaniel**
- Broke out step-by-step guide into different files we can stick in readmes for other projects
- Took out some extra steps we're not using anymore and steps that don't have a guide i.e. boot2docker, AWS CLI and SSH
- Added new URLs to setup guide.

NOTE: the URLs currently in the setup guide will break once pushed to master because they reference files only on this branch. I can change now or later, whichever.

### CODE REVIEW

**Comment by anniedaniel** [comment]
Rework landing readme to point to new guides

**Comment by anniedaniel** [comment]
Get rid of code linting guide

**Comment by anniedaniel** [comment]
Link to better git guide. Any suggestions?

**Comment by rdmurphy** [comment]
My notes, stream of consciousness style:
The code linting guide is old and out of date — it's probably doing more harm than good without a re-write.

The same for the `build-a-graphic` guide. It's out-of-date to the point of harmful. I say cut it for now.

I think the `git` guide in the root of the project is a bit long and hard to follow. Either we super-tighten that up, or we link out to other guides. The interactive git tutorial at https://try.github.io/ is nice. Github also has a quick "what is Github" walk-through as well. https://guides.github.com/activities/hello-world/. The one in the folder is good.

I'd say we drop Visual Studio Code from the editors guide. No one here uses it, and I don't wanna point something in that direction and then have to debug that blind. We can also drop all mention of Sublime Text 2 vs. 3 in all of this — 3 is the default now. The Sublime Text homepage links to 3 now.

I'd like to drop any mention of Python 2 in the standard guides, and set up a separate "so you have to use Python 2, huh?" guide. I think it'll streamline things quite a bit.

I'd rather just direct folks to the Postgres.app guide on its homepage to install than have it duplicated in our guide. http://postgresapp.com/ No need to leave in the mention of us using it instead of Homebrew.

The Node guide is super sparse. Just telling people to install `nvm` isn't enough. I'll probably take a pass at that.

We should bring over the `xcode-select --install` line to the Homebrew guide. That's why you gotta do it.

We should also just drop the `.bash_profile` guide. That's no longer necessary.

One last thing - I get why we broke them all out, but some of these guides are so tiny (or barely guides at all and just link out to a site) that I kind of wish they were one file again. Or maybe one "setup" README, but the guides that are longer have links out to separate pages. It's a lot to click through right now.
